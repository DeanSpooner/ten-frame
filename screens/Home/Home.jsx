import { Text } from "react-native";
import homeStyles from "./Home.styles";

const Home = () => {
  return (
    <>
      <Text style={homeStyles.text}>Welcome to the Ten Frame app!</Text>
    </>
  );
};

export default Home;
