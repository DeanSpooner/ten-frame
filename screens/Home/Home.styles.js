import { StyleSheet } from "react-native";

export default homeStyles = StyleSheet.create({
  cell: {
    width: "100px",
    height: "100px",
  },
  text: {
    color: "#fff",
    fontSize: 30,
    textAlign: "center",
    marginHorizontal: 5,
  },
  textInput: {
    height: 60,
    margin: 12,
    width: "100%",
    padding: 10,
    fontSize: 40,
    textAlign: "center",
  },
  inputContainer: {
    backgroundColor: "#fff",
    width: "25%",
    height: 60,
    marginVertical: 10,
    borderRadius: 10,
    alignItems: "center",
    justifyContent: "center",
    flexDirection: "column",
  },
});
