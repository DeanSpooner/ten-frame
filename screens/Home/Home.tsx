import React from "react";
import { Text } from "react-native";
import homeStyles from "./Home.styles";

const Home: React.FC<{}> = () => {
  return (
    <>
      <Text style={homeStyles.text}>Welcome to the Ten Frame app!</Text>
    </>
  );
};

export default Home;
