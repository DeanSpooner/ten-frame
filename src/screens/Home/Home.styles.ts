import { StyleSheet } from "react-native";

const homeStyles = StyleSheet.create({
  text: {
    color: "#fff",
    fontSize: 30,
    textAlign: "center",
    marginTop: 10,
  },
  menuItemContainer: {
    zIndex: 102,
    elevation: 102,
    backgroundColor: "rgba(0,255,205,0.4)",
    borderRadius: 12,
    alignItems: "center",
    marginTop: 20,
  },
  menuItem: {
    borderRadius: 12,
    color: "#fff",
    padding: 16,
    fontSize: 30,
    textAlign: "center",
    justifyContent: "center",
  },
});

export default homeStyles;
