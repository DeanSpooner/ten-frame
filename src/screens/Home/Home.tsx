import React from "react";
import { Text, TouchableOpacity, View } from "react-native";
import { screens } from "../../data/screen";
import homeStyles from "./Home.styles";

interface HomeProps {
  setScreen: React.Dispatch<React.SetStateAction<string>>;
}

const Home: React.FC<HomeProps> = ({ setScreen }) => {
  return (
    <>
      <Text style={homeStyles.text}>Welcome to the Ten Frame app!</Text>
      <Text style={[homeStyles.text, { fontStyle: "italic" }]}>
        Choose a mode to get started:
      </Text>
      <View>
        {screens.map(
          (screen) =>
            screen.name !== "home" && (
              <TouchableOpacity
                onPress={() => setScreen(screen.name)}
                style={homeStyles.menuItemContainer}
                key={screen.name}
              >
                <Text style={homeStyles.menuItem}>{screen.text}</Text>
              </TouchableOpacity>
            )
        )}
      </View>
    </>
  );
};

export default Home;
