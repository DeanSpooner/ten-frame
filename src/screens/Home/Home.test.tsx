import React from "react";
import renderer, { ReactTestRendererJSON } from "react-test-renderer";
import Home from "./Home";

describe("<Home />", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Home
          setScreen={function (value: React.SetStateAction<string>): void {
            throw new Error("Function not implemented.");
          }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("contains 3 direct child components ", () => {
    const tree = renderer
      .create(
        <Home
          setScreen={function (value: React.SetStateAction<string>): void {
            throw new Error("Function not implemented.");
          }}
        />
      )
      .toJSON() as ReactTestRendererJSON[];
    expect(tree?.length).toBe(3);
  });
});
