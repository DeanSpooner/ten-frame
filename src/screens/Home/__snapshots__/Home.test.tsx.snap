// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders correctly 1`] = `
[
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 30,
        "marginTop": 10,
        "textAlign": "center",
      }
    }
  >
    Welcome to the Ten Frame app!
  </Text>,
  <Text
    style={
      [
        {
          "color": "#fff",
          "fontSize": 30,
          "marginTop": 10,
          "textAlign": "center",
        },
        {
          "fontStyle": "italic",
        },
      ]
    }
  >
    Choose a mode to get started:
  </Text>,
  <View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(0,255,205,0.4)",
          "borderRadius": 12,
          "elevation": 102,
          "marginTop": 20,
          "opacity": 1,
          "zIndex": 102,
        }
      }
    >
      <Text
        style={
          {
            "borderRadius": 12,
            "color": "#fff",
            "fontSize": 30,
            "justifyContent": "center",
            "padding": 16,
            "textAlign": "center",
          }
        }
      >
        Ten Frame - Input
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(0,255,205,0.4)",
          "borderRadius": 12,
          "elevation": 102,
          "marginTop": 20,
          "opacity": 1,
          "zIndex": 102,
        }
      }
    >
      <Text
        style={
          {
            "borderRadius": 12,
            "color": "#fff",
            "fontSize": 30,
            "justifyContent": "center",
            "padding": 16,
            "textAlign": "center",
          }
        }
      >
        Ten Frame - Touch
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(0,255,205,0.4)",
          "borderRadius": 12,
          "elevation": 102,
          "marginTop": 20,
          "opacity": 1,
          "zIndex": 102,
        }
      }
    >
      <Text
        style={
          {
            "borderRadius": 12,
            "color": "#fff",
            "fontSize": 30,
            "justifyContent": "center",
            "padding": 16,
            "textAlign": "center",
          }
        }
      >
        Ten Frame - Game
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(0,255,205,0.4)",
          "borderRadius": 12,
          "elevation": 102,
          "marginTop": 20,
          "opacity": 1,
          "zIndex": 102,
        }
      }
    >
      <Text
        style={
          {
            "borderRadius": 12,
            "color": "#fff",
            "fontSize": 30,
            "justifyContent": "center",
            "padding": 16,
            "textAlign": "center",
          }
        }
      >
        About
      </Text>
    </View>
  </View>,
]
`;
